### PR TITLE
GRN2-155: Begin preparing for removal of Twitter accounts

### PR DESCRIPTION
**Pull request by shawn-higgins1**

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -51,6 +51,10 @@ def omniauth
       @auth = request.env['omniauth.auth']
       @user_exists = check_user_exists
 
+      if !@user_exists && @auth['provider'] == "twitter"
+        return redirect_to root_path, flash: { alert: I18n.t("registration.deprecated.twitter_signup") }
+      end
```
Comment (farhatahmad) [inline app/controllers/sessions_controller.rb]: This can be done in 1 line (see the line below it)

File: app/helpers/sessions_helper.rb [hunk 2]
```diff
@@ -19,6 +19,24 @@
 module SessionsHelper
   # Logs a user into GreenLight.
   def login(user)
+    if !session["old_twitter_user_id"].nil? && user.provider != "twitter"
+      old_user = User.find(session["old_twitter_user_id"])
+
```
Comment (farhatahmad) [inline app/helpers/sessions_helper.rb]: I would suggest moving this code into its own private method

File: app/controllers/sessions_controller.rb (hunk 1 above)
Comment (shawn-higgins1) [reply app/controllers/sessions_controller.rb]: Then it goes over the line column limit

Comment (farhatahmad) [reply app/controllers/sessions_controller.rb]: Hm. Good point. In that case I would leave it as is

**Pull request merged**